### DISCUSSION TITLE
How to show relationship from array in table

### DISCUSSION THREAD

**Discussion by christmex** [filamentphp/filament#7520] [Help] [answered]
Hi guys, I have a question about how to show the relationship from an array, so I have a Book Model and Author Model
here's my Book Model
```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Book extends Model
{
    use HasFactory;

    protected $guarded = [];

    protected $casts = [
        'book_authors' => 'array',
    ];

    public function authors(){
        return $this->belongsTo(Author::class,'book_authors','id');
    }
}

```

My Author Model
```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Author extends Model
{
    use HasFactory;

    protected $guarded = [];
}

```

And inside my BookResource, I have a function that looks like this to show 
```php
    public static function table(Table $table): Table
    {
          return $table
          ->columns([
                Tables\Columns\TextColumn::make('authors')
                ->searchable(),
          ])
          //... another code
    }
```
the result shows me this
<img width="432" alt="image" src="https://github.com/filamentphp/filament/assets/34405837/e572a071-9c54-4200-87e6-ca87d664b564">

In my opinion I can access the author_name with dot, so it will look like this ``` Tables\Columns\TextColumn::make('authors.author_name') ```
but then I got this error
<img width="374" alt="image" src="https://github.com/filamentphp/filament/assets/34405837/a4021970-587d-4747-9a24-b38f2bb97c01">

### **What I want**
<img width="156" alt="image" src="https://github.com/filamentphp/filament/assets/34405837/691749c9-f00f-4915-9664-66feb4075255">

How can I achieve this look? is it possible, or did I do something wrong? Thanks for your time.

**Comment by danharrin** [comment]
`getStateUsing(fn ($record) => collect($record->authors)->pluck('author_name')->all())`

**Comment by christmex** [comment]
Hi @danharrin thank you for answering my question

Sorry I'm new to filament, I did something like this
```php
Tables\Columns\TextColumn::make('book_authors')
      ->getStateUsing(fn ($record) => collect($record->authors)->pluck('author_name')->all())
      ->searchable()
```
or even this
```php
Tables\Columns\TextColumn::make('authors')
    ->getStateUsing(fn ($record) => collect($record->authors)->pluck('author_name')->all())
    ->searchable()
```
the result same look like this
<img width="73" alt="image" src="https://github.com/filamentphp/filament/assets/34405837/5aaf0fd7-05b8-4e52-acc9-ac29652230fb">

I am also trying to find a way and Here's what I do to show them, but when I search it doesn't work really well
```php
Tables\Columns\TextColumn::make('book_name')
    ->description(function(Book $record){
        $authors = Author::whereIn('id', $record->book_authors)->pluck('author_name')->toArray();
        return implode(', ', $authors);
    })
    ->searchable(),
```

did I miss something? I'm sorry.

**Comment by danharrin** [comment, marked as answer]
I've just realised youre trying to use this as an actual relationship. Eloquent doesnt really work like that, you should use a separate table

**Reply by christmex** [reply]
@danharrin thank you so much for letting me know 🙏🏽, I understand.
